### PR TITLE
Made improvements to the beacon PR

### PR DESCRIPTION
**Pull request by JavierLeon9966**

### CODE REVIEW

File: src/block/Beacon.php [hunk 1]
```diff
@@ -32,9 +32,56 @@
...
+		}
+	}
+
+	public function getPrimaryEffect() : int{
+		return $this->primaryEffect;
+	}
+
+	/** @return $this */
+	public function setPrimaryEffect(int $primaryEffect) : self{
+		$this->primaryEffect = $primaryEffect;
```
Comment (ShockedPlot7560) [inline src/block/Beacon.php:92]: Perhaps it would be necessary to check the effects sent, as stated in the channel?

File: src/block/Beacon.php [hunk 2]
```diff
@@ -32,9 +32,56 @@
...
+		return $this;
+	}
+
+	public function getSecondaryEffect() : int{
+		return $this->secondaryEffect;
+	}
+
+	/** @return $this */
+	public function setSecondaryEffect(int $secondaryEffect) : self{
+		$this->secondaryEffect = $secondaryEffect;
```
Comment (ShockedPlot7560) [inline src/block/Beacon.php:108]: Same as primaryEffect

File: src/block/Beacon.php [hunk 3]
```diff
@@ -73,18 +120,16 @@ public function onScheduledUpdate() : void{
...
-			foreach($this->position->getWorld()->getPlayers() as $player){
-				if($player->getPosition()->distance($this->position) <= $radius){
-					if($primaryE === $secondaryE){
-						$effect = EffectIdMap::getInstance()->fromId($secondaryE);
-						if($effect instanceof Effect){
-							$player->getEffects()->add(new EffectInstance($effect, $effectDuration * 20, 1));
-						}
-						break;
+			if($primaryE === $secondaryE){
+				foreach($this->position->getWorld()->getPlayers() as $player){
```
Comment (ShockedPlot7560) [inline src/block/Beacon.php]: To improve the performance, as some people have said, I think you should also use getNearbyPlayers

File: src/block/Beacon.php [hunk 4]
```diff
@@ -73,18 +120,16 @@ public function onScheduledUpdate() : void{
...
+					if($player->getPosition()->distance($this->position) <= $radius){
+						$player->getEffects()->add(new EffectInstance($effect, $effectDuration * 20, 1));
 					}
-					foreach([$primaryE, $secondaryE] as $effectId){
-						if($effectId !== 0){
-							$effect = EffectIdMap::getInstance()->fromId($effectId);
+				}
+			}else{
+				foreach($this->position->getWorld()->getPlayers() as $player){
```
Comment (ShockedPlot7560) [inline src/block/Beacon.php]: same as above

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
some modification, and I think we'll be good, I'll take care of it in a while, or you can start

**Comment by JavierLeon9966** [comment]
Done

**Pull request merged**